src/layer.rs

```diff
@@ -97,14 +155,40 @@ impl<T: api::Tracer + 'static> OpentelemetryLayer<T> {
     /// let tracer = provider.get_tracer("component-name");
     ///
     /// // Create a layer with the configured tracer
-    /// let _layer = OpentelemetryLayer::with_tracer(tracer);
+    /// let otel_layer = OpenTelemetryLayer::with_tracer(tracer);
+    ///
+    /// // Use the tracing subscriber `Registry`, or any other subscriber
+    /// // that impls `LookupSpan`
+    /// let _subscriber = otel_layer.with_subscriber(Registry::default());
```

Comment:
nit, take it or leave it: generally, I find using `SubscriberExt::with` a little nicer looking than `Layer::with_subscriber`, and tend to prefer that in examples:
```suggestion
    /// let _subscriber = Registry::default())
    ///     .with(otel_layer);
```

Reply:
also, take it or leave it, but it might be better to show the subscriber actually being used here, rather than using `_`...maybe end the example with `tracing::subscriber::set_global_default` or something? I don't want new users to think that just _constructing_ a subscriber is all they need to do to get traces working.

Just a thought!

Reply:
Yeah good point. Better to show more complete examples, thanks.